### ISSUE TITLE
New endpoint request

### ISSUE THREAD

**Issue by Feldspar5** [namc-utah/namc-bug-db#221]
Phillip - We are requesting an endpoint that outputs the following for all sampleids where predictor status is not valid:

1. SampleID
2. SiteID
3. Sample date

We can place this in code, but would be much more efficient (we think) if we could access directly.  Obviously this produces a list of samples that we can then run through the predictor extraction function.

Thanks

**Comment by Feldspar5** [comment]
Or maybe this could just simply be a list of sampleids (dropping 2 and 3 above).

**Comment by philipbaileynar** [comment]
This is easy. I can do this. 

It suggests to me that you intend to periodically run the predictor calculation process for "all samples that have missing or invalid values". 

This is fine, but it's different than I had anticipated though. I was thinking you would calculate predictors for a particular box or project, looping over all the relevant samples, calculating predictors where needed. 

Both approaches are valid. I will implement the new endpoint to give you options.

**Comment by Feldspar5** [comment]
Yes that is how we usually run this process.  However, David has steered us towards being able to do that AND implementing the process (ultimately) automatically.  In other words, the function scans the db for completed samples with incomplete (invalid) predictor values and then extracts and populates those.
